src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -122,25 +122,49 @@ public function match(\$rawPathinfo)
 
 $code
 
-        throw \$allow ? new MethodNotAllowedException(array_keys(\$allow)) : new ResourceNotFoundException();
-    }
 EOF;
+
+        if ($this->supportsRedirections) {
+            return <<<'EOF'
+    public function match($pathinfo)
+    {
+        $allow = array();
+        if ($ret = $this->doMatch($pathinfo, $allow)) {
+            return $ret;
+        }
+        if ('/' !== $pathinfo && in_array($this->context->getMethod(), array('HEAD', 'GET'), true)) {
+            $pathinfo = '/' !== $pathinfo[-1] ? $pathinfo.'/' : substr($pathinfo, 0, -1);
```

Comment:
The routing component does not dintinguish between `/` and encoded `%2F`. So I think the redirection should handle both the same as well. That means we would need to rawurldecode in match and not only in doMatch.

E.g. if you try to match `http://symfony.com/blog/category/documentation%2F` -> fail because wrong traling slash.
It should then try to match `http://symfony.com/blog/category/documentation` and not `http://symfony.com/blog/category/documentation%2F/` as now.

Reply:
That's not how RedirecteableUrlMatcher behaves. Tests also expect the original encoding be preserved. In the end, I think we should not try harder handling these edge cases. The goal of trailing slash redirections is helping navigating via the browser bar. And webservers & browsers already normalize this for us most of the time. Let's keep things simple.

Reply:
Fine for me.